### PR TITLE
Add support for aws kinesis as a dispatcher

### PR DESCRIPTION
**Pull request by aaronpkahn**

### CODE REVIEW

File: test/integration/Dockerfile [hunk 1]
```diff
@@ -1,8 +1,8 @@
 FROM golang:1.20.3
```
Comment (vmallet) [inline test/integration/Dockerfile:1]:
```suggestion
FROM golang:1.20.5-bullseye
```

**Review by vmallet — COMMENTED** [review]
The go binaries should be built on `golang:1.20.5-bullseye`, otherwise it's picking up `bookworm` which has a newer libc than distroless.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -17,13 +17,13 @@ jobs:
...
+      run: make integration
+
     - name: Format
       run: |
         make format
         git diff --exit-code
 
     - name: Test
       run: make test
-
```
Comment (agbpatro) [inline .github/workflows/build.yml:27]: revert this change

File: datastore/kinesis/kinesis.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+		PartitionKey: aws.String(entry.Vin),
+	}
+	kinesisRecordOutput, err := p.kinesis.PutRecord(kinesisRecord)
+	if err != nil {
+		p.logger.Errorf("kinesis_err: %v", err)
+		metrics.StatsIncrement(p.statsCollector, "kinesis_err", 1, map[string]string{"record_type": entry.TxType})
+		return
+	}
+
+	p.logger.Debugf("kinesis_publish vin=%s,type=%s,txid=%s,shard_id=%s,sequence_number=%s", entry.Vin, entry.TxType, entry.Txid, *kinesisRecordOutput.ShardId, *kinesisRecordOutput.SequenceNumber)
```
Comment (agbpatro) [inline datastore/kinesis/kinesis.go:72]: can we remove this?

File: test/integration/config.json [hunk 4]
```diff
@@ -16,10 +16,14 @@
...
     },
+    "kinesis": {
+        "override_host": "http://kinesis:4567",
+        "max_retries" : 3
+    },
     "monitoring": {
         "prometheus_metrics_port": 9090,
         "profiler_port": 4269,
-        "profiling_path": "/tmp/tesla/tesla-telemetry/profile"
+        "profiling_path": "/tmp/tesla/fleet-telemetry/profile"
```
Comment (agbpatro) [inline test/integration/config.json:26]: nit: should this be `/tmp/fleet-telemetry/profile`?

**Review by agbpatro — APPROVED** [review]

File: datastore/kinesis/kinesis.go (hunk 3 above)
Comment (aaronpkahn) [reply datastore/kinesis/kinesis.go:72]: Seems useful for debugging.

File: test/integration/config.json (hunk 4 above)
Comment (aaronpkahn) [reply test/integration/config.json]: updated
